application/models/SuppliesModel.php

```diff
@@ -9,166 +9,79 @@
+            ['field' => 'quantityOnHand', 'label' => 'Supply quantity', 'rules' => 'required|integer']
+        ];
+        return $config;
+    }
 
-    public function details($id)
+    // Return all records as an array of objects
+    function all()
     {
-        foreach ($this->data as $supply) {
-            if ($supply['id'] == $id) {
-                return $supply;
-            }
-        }
+            $this->rest->initialize(array('server' => REST_SERVER));
+            $this->rest->option(CURLOPT_PORT, REST_PORT);
+            return $this->rest->get('/maintenance');
+    }
 
-        return null;
+        // Retrieve an existing DB record as an object
+    function get($key, $key2 = null)
+    {
+            $this->rest->initialize(array('server' => REST_SERVER));
+            $this->rest->option(CURLOPT_PORT, REST_PORT);
+            return $this->rest->get('/maintenance/item/id/' . $key);
+    }
+    // Create a new data object.
+    // Only use this method if intending to create an empty record and then
+    // populate it.
+    function create()
+    {
+        $names = ['id','name','description','price','picture','category'];
+        $object = new StdClass;
+        foreach ($names as $name)
+            $object->$name = "";
+        return $object;
+    }
+    // Delete a record from the DB
+    function delete($key, $key2 = null)
+    {
+            $this->rest->initialize(array('server' => REST_SERVER));
+            $this->rest->option(CURLOPT_PORT, REST_PORT);
+            return $this->rest->delete('/maintenance/item/id/' . $key);
     }
 
-        // Add a supply item to the data.
-        public function addSupply($supply) {
-            array_push($this->data, $supply);
-        }
+    // Determine if a key exists
+    function exists($key, $key2 = null)
+    {
+            $this->rest->initialize(array('server' => REST_SERVER));
+            $this->rest->option(CURLOPT_PORT, REST_PORT);
+            $result = $this->rest->get('/maintenance/item/id/' . $key);
+            return ! empty($result);
+    }
+    // Update a record in the DB
+    function update($record)
+    {
+            $this->rest->initialize(array('server' => REST_SERVER));
+            $this->rest->option(CURLOPT_PORT, REST_PORT);
+            $retrieved = $this->rest->put('/maintenance/item/id/' . $record['code'], $record);
+    }
 
-        // Delete a supply item from the data.
-        public function deleteSupply($code) {
-            foreach ($this->data as $supply) {
-                if ($supply['code'] == $code) {
-                    unset($this->data[$code]);
-                }
-            }
-        }
+    // Add a record to the DB
+    function add($record)
+    {
+            $this->rest->initialize(array('server' => REST_SERVER));
+            $this->rest->option(CURLOPT_PORT, REST_PORT);
+            $retrieved = $this->rest->post('/maintenance/item/id/' . $record['code'], $record);
+    }
```

Comment:
$record[code] should be $record[id] I think. I kept the IDs that were put in from the first assignment. I think it makes things easier using a number instead of a string with spaces.